pkg/scheduler/internal/cache/cache.go

```diff
@@ -537,6 +548,7 @@ func (cache *schedulerCache) UpdateNode(oldNode, newNode *v1.Node) error {
 	if !ok {
 		n = newNodeInfoListItem(schedulernodeinfo.NewNodeInfo())
 		cache.nodes[newNode.Name] = n
+		metrics.CacheSize.WithLabelValues("nodes").Inc()
```

Comment:
Why are we incrementing the metric when a node is updated? this is an example where we could get things wrong with Inc/Dec, but using Set would just be redundant but not invalid.